tests/PHPStan/Analyser/data/bug-4650.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace Bug4213;
+
+use function PHPStan\Analyser\assertNativeType;
+use function PHPStan\Analyser\assertType;
+
+/**
+ * @phpstan-param non-empty-array<string|int> $idx
+ */
+function (array $idx): void {
+	assertType('bool', $idx === []);
+	assertNativeType('bool', $idx === []);
+	assertType('bool', $idx !== []);
```

Comment:
This should be true

Reply:
Then it fails
```
1) PHPStan\Analyser\NodeScopeResolverTest::testFileAsserts with data set "/Users/Ruud/opensource/phpstan-src/tests/PHPStan/Analyser/data/bug-4650.php:12" ('type', '/Users/Ruud/opensource/phpsta...50.php', PHPStan\Type\Constant\ConstantStringType Object (...), PHPStan\Type\BooleanType Object (), 12)
Expected type false, got type bool in /Users/Ruud/opensource/phpstan-src/tests/PHPStan/Analyser/data/bug-4650.php on line 12.
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'false'
+'bool'

/Users/Ruud/opensource/phpstan-src/tests/PHPStan/Analyser/NodeScopeResolverTest.php:11333

2) PHPStan\Analyser\NodeScopeResolverTest::testFileAsserts with data set "/Users/Ruud/opensource/phpstan-src/tests/PHPStan/Analyser/data/bug-4650.php:14" ('type', '/Users/Ruud/opensource/phpsta...50.php', PHPStan\Type\Constant\ConstantStringType Object (...), PHPStan\Type\BooleanType Object (), 14)
Expected type true, got type bool in /Users/Ruud/opensource/phpstan-src/tests/PHPStan/Analyser/data/bug-4650.php on line 14.
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'true'
+'bool'
```

It's also why I think that the test is not good enough. Shouldn't it be tested with `treatPhpDocTypesAsCertain: false`?